src/textual/widgets/_list_view.py

```diff
@@ -216,6 +216,53 @@ def clear(self) -> AwaitRemove:
         self.index = None
         return await_remove
 
+    def insert(self, index: int, items: Iterable[ListItem]) -> AwaitMount:
+        """Insert new ListItem(s) to specified index.
+
+        Args:
+            index: index to insert new ListItem.
+            items: The ListItems to insert.
+
+        Returns:
+            An awaitable that yields control to the event loop
+                until the DOM has been updated with the new child item. 
+        """
+        await_mount = self.mount(*items, before=index)
+        return await_mount
+
+    def pop(self, index: Optional[int]=None) -> AwaitRemove:
+        """Remove last ListItem from ListView or
+           Remove ListItem from ListView by index
+
+        Args:
+            index: index of ListItem to remove from ListView
+
+        Returns:
+            An awaitable that yields control to the event loop until
+                the DOM has been updated to reflect item being removed.
+        """
+        if index is None:
+            await_remove = self.query("ListItem").last().remove()
+        else:
+            await_remove = self.query("ListItem")[index].remove()
+        return await_remove
+
+    def remove_items(self, indices: Iterable[int]) -> AwaitRemove:
+        """Remove ListItems from ListView by indices
+
+        Args:
+            indices: index(s) of ListItems to remove from ListView
+
+        Returns:
+            An awaitable object that waits for the direct children to be removed. 
+        """
+        items_to_remove = []
+        for index in indices:
+            items_to_remove.append(self.query("ListItem")[index])
```

Comment:
You could cache that query. Otherwise it would be performed once per loop.

Reply:
please have a review